src/validation.cpp

```diff
@@ -4188,7 +4188,7 @@ static bool ContextualCheckBlockHeader(const CBlockHeader& block, BlockValidatio
         // Check timestamp for the first block of each difficulty adjustment
         // interval, except the genesis block.
         if (nHeight % consensusParams.DifficultyAdjustmentInterval() == 0) {
-            if (block.GetBlockTime() < pindexPrev->GetBlockTime() - 60 * 60 * 2) {
+            if (block.GetBlockTime() < pindexPrev->GetBlockTime() - MAX_FUTURE_BLOCK_TIME) {
```

Comment:
35ad61a6bde14a74cc07dc8ecf3bb1cd49590e46: let's define a separate constant and make it clear these are different but related:

```cpp
/**
 * Maximum number of seconds that the timestamp of the first
 * block of a difficulty adjustment period is allowed to
 * be earlier than the last block of the previous period.
 */
static constexpr int64_t MAX_TIMEWARP{MAX_FUTURE_BLOCK_TIME};

/**
 * If the timestamp of the last block in a difficulty period is set
 * MAX_FUTURE_BLOCK_TIME seconds in the future, an honest miner should
 * be able to mine the first block using the current time.
 */
static_assert(MAX_FUTURE_BLOCK_TIME <= MAX_TIMEWARP);
```

See ongoing discussion in https://github.com/bitcoin/bips/pull/1658

Reply:
And see #30614.

Reply:
> let's define a separate constant

done

Reply:
> And see https://github.com/bitcoin/bitcoin/issues/30614, but that should probably be its own followup.

Agree, it seems that we currently have the safer option given we don't really know what other mining software is doing (at least that's my feeling). If we do still get consensus on tightening the limit we can do that in another follow-up in the next two weeks but it shouldn't hold up this PR here.